src/main/java/build/buildfarm/server/ByteStreamService.java

```diff
@@ -106,9 +106,20 @@ private void readBlob(
       return;
     }
 
-    responseObserver.onNext(ReadResponse.newBuilder()
-        .setData(blob)
-        .build());
+    while (!blob.isEmpty()) {
+      ByteString chunk;
+      if (blob.size() < DEFAULT_CHUNK_SIZE) {
+        chunk = blob;
+        blob = ByteString.EMPTY;
+      } else {
+        chunk = blob.substring(0, (int) DEFAULT_CHUNK_SIZE);
+        blob = blob.substring((int) DEFAULT_CHUNK_SIZE);
+      }
+      responseObserver.onNext(ReadResponse.newBuilder()
```

Comment:
Duly noted - the interface for the blob itself could be unnecessarily consumptive as well - since the content is guaranteed by the digest, and we can error the stream safely, I should only be requesting a chunk at a time from the instance.